astronomer/providers/google/cloud/operators/bigquery.py

```diff
@@ -124,18 +124,23 @@ def execute(self, context: Context) -> None:  # noqa: D102
 
         self.job_id = job.job_id
         context["ti"].xcom_push(key="job_id", value=self.job_id)
-        self.defer(
-            timeout=self.execution_timeout,
-            trigger=BigQueryInsertJobTrigger(
-                conn_id=self.gcp_conn_id,
-                job_id=self.job_id,
-                project_id=self.project_id,
-                impersonation_chain=self.impersonation_chain,
-                poll_interval=self.poll_interval,
-                **kwargs,
-            ),
-            method_name="execute_complete",
-        )
+        if job.running():
+            self.defer(
+                timeout=self.execution_timeout,
+                trigger=BigQueryInsertJobTrigger(
+                    conn_id=self.gcp_conn_id,
+                    job_id=self.job_id,
+                    project_id=self.project_id,
+                    impersonation_chain=self.impersonation_chain,
+                    poll_interval=self.poll_interval,
+                    **kwargs,
+                ),
+                method_name="execute_complete",
+            )
+        else:
+            job.result()
```

Comment:
I was looking at `BigQueryInsertJobOperator` implementation for the sync version they do `job.result()`  so here if somehow else block gets executed it would be the same as the sync operator??

Reply:
Yes, we're doing the same thing without waiting and retrying, as the job is not in running status. If it is, we'll defer it

Reply:
yes, but that should be okay. We know here that the job is not running anymore, has reached a terminal state and would like to proceed ahead without having an overhead of the deferral.